specification/securitycenter/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json

```diff
@@ -0,0 +1,1955 @@
+                    "items": {
+                        "$ref": "#/definitions/DiscoveredSecuritySolution"
+                    }
+                }
+            }
+        },
+        "DiscoveredSecuritySolution": {
+            "type": "object",
+            "required": [
+                "properties",
+                "id",
+                "name",
+                "type",
+                "location"
+            ],
+            "properties": {
+                "properties": {
+                    "required": [
+                        "securityFamily",
+                        "offer",
+                        "publisher",
+                        "sku"
+                    ],
+                    "type": "object",
+                    "properties": {
+                        "securityFamily": {
+                            "$ref": "#/definitions/securityFamily"
+                        },
+                        "offer": {
+                            "type": "string",
+                            "description": "The security solutions' image offer"
+                        },
+                        "publisher": {
+                            "type": "string",
+                            "description": "The security solutions' image publisher"
+                        },
+                        "sku": {
+                            "type": "string",
+                            "description": "The security solutions' image sku"
+                        }
+                    }
+                },
+                "type": {
+                    "type": "string",
+                    "description": "Type of the resource",
+                    "enum": [
+                        "Microsoft.Security/locations/discoveredSecuritySolution"
+                    ]
+                }
+            },
+            "allOf": [{
+                "$ref": "#/definitions/Resource"
+            },
+            {
+                "$ref": "#/definitions/Location"
+            }]
+        },
+        "securityFamily": {
+            "type": "string",
+            "description": "The security family of the discovered solution",
+            "enum": [
+                "Waf",
+                "Ngfw",
+                "SaasWaf",
+                "Va"
+            ]
+        },
+        "JitNetworkAccessPoliciesList": {
+            "type": "object",
+            "properties": {
+                "value": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/JitNetworkAccessPolicy"
+                    }
+                }
+            }
+        },
+        "JitNetworkAccessPolicy": {
+            "type": "object",
+            "required": [
+                "kind",
+                "properties",
+                "id",
+                "name",
+                "type",
+                "location"
+            ],
+            "properties": {
+                "properties": {

```

Comment:
Please do not define complex models inline. It causes a lot of problem during model validation. Moreover it is not easy to read for a human. Swagger "properties" and ARM "properties" are interleaved. Take a look at [this example](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/Microsoft.Storage/stable/2017-10-01/storage.json#L1390)
/cc @sergey-shandar - Please take a note of this, to call out such patterns in the future.